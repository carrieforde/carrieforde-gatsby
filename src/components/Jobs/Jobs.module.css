.jobPanel p {
  margin-top: 0;
}
