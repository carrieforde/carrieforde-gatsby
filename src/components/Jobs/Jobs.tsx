import Tabs from 'components/Tabs/Tabs';
import { Tab } from 'components/Tabs/Tabs.interface';
import { graphql, useStaticQuery } from 'gatsby';
import { MDXRenderer } from 'gatsby-plugin-mdx';
import { Edge } from 'interfaces/edge.interface';
import React from 'react';
import { useMemo } from 'react';
import { jobPanel } from './Jobs.module.css';

const Jobs: React.FC = () => {
  const { allMdx } = useStaticQuery(graphql`
    query {
      allMdx(
        filter: { fileAbsolutePath: { regex: "/jobs/" } }
        sort: { fields: [frontmatter___date], order: DESC }
      ) {
        edges {
          node {
            frontmatter {
              company
              date
            }
            body
          }
        }
      }
    }
  `);

  const tabData = useMemo(() => {
    const { edges } = allMdx;

    return edges.map((e: Edge) => ({
      title: e.node.frontmatter.company,
      content: <MDXRenderer>{e.node.body}</MDXRenderer>,
    })) as Tab[];
  }, []);

  return <Tabs tabItems={tabData} ariaLabel="Jobs" panelClass={jobPanel} />;
};

export default Jobs;
