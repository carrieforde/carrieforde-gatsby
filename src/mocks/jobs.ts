export const jobs = {
  allMdx: {
    edges: [
      {
        node: {
          frontmatter: {
            date: '2018-06-11',
          },
          body: 'var _excluded = ["components"];\n\nfunction _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }\n\nfunction _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }\n\nfunction _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }\n\n/* @jsxRuntime classic */\n\n/* @jsx mdx */\nvar _frontmatter = {\n  "company": "Freedom Financial Network",\n  "date": "2018-06-11"\n};\n\nvar makeShortcode = function makeShortcode(name) {\n  return function MDXDefaultShortcode(props) {\n    console.warn("Component " + name + " was not imported, exported, or provided by MDXProvider as global scope");\n    return mdx("div", props);\n  };\n};\n\nvar Heading = makeShortcode("Heading");\nvar TextWithSeparator = makeShortcode("TextWithSeparator");\nvar layoutProps = {\n  _frontmatter: _frontmatter\n};\nvar MDXLayout = "wrapper";\nreturn function MDXContent(_ref) {\n  var components = _ref.components,\n      props = _objectWithoutProperties(_ref, _excluded);\n\n  return mdx(MDXLayout, _extends({}, layoutProps, props, {\n    components: components,\n    mdxType: "MDXLayout"\n  }), mdx(Heading, {\n    level: "h3",\n    weight: 300,\n    mdxType: "Heading"\n  }, "Senior Software Engineer @ Freedom Financial Network"), mdx(TextWithSeparator, {\n    text1: "San Mateo, CA",\n    text2: "June 2018 - Present",\n    mdxType: "TextWithSeparator"\n  }), mdx("p", null, "Lead architecture and development of company WordPress properties, while working to improve site performance metrics. Develop sophisticated templates and features using JavaScript (including React), Sass, and PHP. Responsible for technical SEO on ", mdx("a", {\n    parentName: "p",\n    "href": "https://www.freedomdebtrelief.com"\n  }, "freedomdebtrelief.com"), "."), mdx("ul", null, mdx("li", {\n    parentName: "ul"\n  }, "Migrated advertorial pages to WordPress from a static site improving speed index of pages by 283%"), mdx("li", {\n    parentName: "ul"\n  }, "Won Best Value to the Business award during company hackathon by extending current online application to help clients enroll in the program online"), mdx("li", {\n    parentName: "ul"\n  }, "Drove early adoption of new WordPress editing experience, implementing usage six months before release into WordPress Core")));\n}\n;\nMDXContent.isMDXComponent = true;',
        },
      },
      {
        node: {
          frontmatter: {
            date: '2016-06-23T00:00:00.000Z',
          },
          body: 'var _excluded = ["components"];\n\nfunction _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }\n\nfunction _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }\n\nfunction _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }\n\n/* @jsxRuntime classic */\n\n/* @jsx mdx */\nvar _frontmatter = {\n  "company": "WebDevStudios",\n  "date": "2016-06-23T00:00:00.000Z"\n};\n\nvar makeShortcode = function makeShortcode(name) {\n  return function MDXDefaultShortcode(props) {\n    console.warn("Component " + name + " was not imported, exported, or provided by MDXProvider as global scope");\n    return mdx("div", props);\n  };\n};\n\nvar Heading = makeShortcode("Heading");\nvar TextWithSeparator = makeShortcode("TextWithSeparator");\nvar layoutProps = {\n  _frontmatter: _frontmatter\n};\nvar MDXLayout = "wrapper";\nreturn function MDXContent(_ref) {\n  var components = _ref.components,\n      props = _objectWithoutProperties(_ref, _excluded);\n\n  return mdx(MDXLayout, _extends({}, layoutProps, props, {\n    components: components,\n    mdxType: "MDXLayout"\n  }), mdx(Heading, {\n    level: "h3",\n    weight: 300,\n    mdxType: "Heading"\n  }, "Senior Front End Developer @ WebDevStudios"), mdx(TextWithSeparator, {\n    text1: "Glenside, PA (remote)",\n    text2: "June 2016 - August 2017",\n    mdxType: "TextWithSeparator"\n  }), mdx("p", null, "Responsible for large, complex front end features, and transforming design mockups into functional WordPress themes using Sass, jQuery, and PHP. Mentored front end developer team members through code reviews and pair programming activities."), mdx("ul", null, mdx("li", {\n    parentName: "ul"\n  }, "Evaluated, improved, and consolidated code for front end components in a large WooCommerce site making components more performant and maintainable"), mdx("li", {\n    parentName: "ul"\n  }, "Managed expectations by communicating potential roadblocks to project managers early avoiding delays in ticket completion"), mdx("li", {\n    parentName: "ul"\n  }, "Influenced policy around theme organization by suggesting developers use template parts to manage smaller, reusable user interface components"), mdx("li", {\n    parentName: "ul"\n  }, "Developed significant portion of scaffolding library within wd_s, WebDevStudios\\u2019 starter theme")));\n}\n;\nMDXContent.isMDXComponent = true;',
        },
      },
      {
        node: {
          frontmatter: {
            date: '2015-07-05T00:00:00.000Z',
          },
          body: 'var _excluded = ["components"];\n\nfunction _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }\n\nfunction _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }\n\nfunction _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }\n\n/* @jsxRuntime classic */\n\n/* @jsx mdx */\nvar _frontmatter = {\n  "company": "MIGHTYminnow",\n  "date": "2015-07-05T00:00:00.000Z"\n};\n\nvar makeShortcode = function makeShortcode(name) {\n  return function MDXDefaultShortcode(props) {\n    console.warn("Component " + name + " was not imported, exported, or provided by MDXProvider as global scope");\n    return mdx("div", props);\n  };\n};\n\nvar Heading = makeShortcode("Heading");\nvar TextWithSeparator = makeShortcode("TextWithSeparator");\nvar layoutProps = {\n  _frontmatter: _frontmatter\n};\nvar MDXLayout = "wrapper";\nreturn function MDXContent(_ref) {\n  var components = _ref.components,\n      props = _objectWithoutProperties(_ref, _excluded);\n\n  return mdx(MDXLayout, _extends({}, layoutProps, props, {\n    components: components,\n    mdxType: "MDXLayout"\n  }), mdx(Heading, {\n    level: "h3",\n    weight: 300,\n    mdxType: "Heading"\n  }, "Front End Developer & Designer @ MIGHTYminnow"), mdx(TextWithSeparator, {\n    text1: "Oakland, CA",\n    text2: "July 2015 - February 2016",\n    mdxType: "TextWithSeparator"\n  }), mdx("p", null, "Designed WordPress websites for nonprofit organizations and small business clients in the San Francisco Bay Area. Gained front end development skills by creating child themes using CSS and PHP. Teaching assistant for live WordPress workshops."), mdx("ul", null, mdx("li", {\n    parentName: "ul"\n  }, "Designed and coded several components for Mm Components, a plugin for adding web components to WordPress websites"), mdx("li", {\n    parentName: "ul"\n  }, "Introduced style guides to maintain consistency when building out client web sites"), mdx("li", {\n    parentName: "ul"\n  }, "Provided design feedback to clients enrolled in Website Weekend classes")));\n}\n;\nMDXContent.isMDXComponent = true;',
        },
      },
      {
        node: {
          frontmatter: {
            date: '2009-10-11T00:00:00.000Z',
          },
          body: 'var _excluded = ["components"];\n\nfunction _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }\n\nfunction _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }\n\nfunction _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }\n\n/* @jsxRuntime classic */\n\n/* @jsx mdx */\nvar _frontmatter = {\n  "company": "Cisco",\n  "date": "2009-10-11T00:00:00.000Z"\n};\n\nvar makeShortcode = function makeShortcode(name) {\n  return function MDXDefaultShortcode(props) {\n    console.warn("Component " + name + " was not imported, exported, or provided by MDXProvider as global scope");\n    return mdx("div", props);\n  };\n};\n\nvar Heading = makeShortcode("Heading");\nvar TextWithSeparator = makeShortcode("TextWithSeparator");\nvar layoutProps = {\n  _frontmatter: _frontmatter\n};\nvar MDXLayout = "wrapper";\nreturn function MDXContent(_ref) {\n  var components = _ref.components,\n      props = _objectWithoutProperties(_ref, _excluded);\n\n  return mdx(MDXLayout, _extends({}, layoutProps, props, {\n    components: components,\n    mdxType: "MDXLayout"\n  }), mdx(Heading, {\n    level: "h3",\n    weight: 300,\n    mdxType: "Heading"\n  }, "Business Process Analyst @ Cisco"), mdx(TextWithSeparator, {\n    text1: "Feltham, UK",\n    text2: "October 2009 - November 2013",\n    mdxType: "TextWithSeparator"\n  }), mdx("p", null, "Managed team of three employees to perform data analysis, reporting, training and tool support for sales operations team of 50 in Europe, the Middle East, Africa, and Russia (EMEAR)."), mdx("ul", null, mdx("li", {\n    parentName: "ul"\n  }, "Identified $250 million in missing renewal opportunity leads during FY10-to-FY11 year-end process; collaborated with IT to regain visibility to opportunity, and with leadership in other geographies to correctly allocate opportunity"), mdx("li", {\n    parentName: "ul"\n  }, "Improved accuracy and efficiency of reporting process by creating a database, saving \\\\$16K by eliminating extra headcount for the position"), mdx("li", {\n    parentName: "ul"\n  }, "Led adoption of sales opportunity management tools in EMEAR market; gathered requirements for sales opportunity management tools")), mdx(Heading, {\n    level: "h3",\n    weight: 300,\n    mdxType: "Heading"\n  }, "Business Analyst @ Cisco"), mdx(TextWithSeparator, {\n    text1: "San Jose, CA",\n    text2: "January 2006 - October 2009",\n    mdxType: "TextWithSeparator"\n  }), mdx("p", null, "Maintained $1.4B renewal base in the Americas; provided renewal opportunity reports to stakeholders. Ensured accurate visibility of renewal opportunities to stakeholders in the sales organization."), mdx("ul", null, mdx("li", {\n    parentName: "ul"\n  }, "Mentored and trained contractor team members; shared best practices for renewal base management and reporting"), mdx("li", {\n    parentName: "ul"\n  }, "Performed root cause analysis to identify issues with renewal opportunity management tools, and worked with global IT for quick resolution to problems")));\n}\n;\nMDXContent.isMDXComponent = true;',
        },
      },
    ],
  },
};
