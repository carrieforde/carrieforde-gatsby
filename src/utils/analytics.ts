interface GtmProps {
  event: string;
  label: string;
  category?: string;
}

export function gtm(props: GtmProps): void {
  if (window.dataLayer) {
    window.dataLayer.push(props);
  }
}
