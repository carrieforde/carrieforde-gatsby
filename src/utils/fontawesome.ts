import { library } from '@fortawesome/fontawesome-svg-core';
import {
  faInfoCircle,
  faLightbulbExclamation,
  faExclamationCircle,
  faExclamationTriangle,
} from '@fortawesome/pro-light-svg-icons';

library.add(
  faInfoCircle,
  faLightbulbExclamation,
  faExclamationCircle,
  faExclamationTriangle
);
