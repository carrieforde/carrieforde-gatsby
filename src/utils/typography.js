import Typography from "typography"
// import CodePlugin from "typography-plugin-code"

const typography = new Typography({
  baseFontSize: "18px",
  baseLineHeight: 1.75,
  googleFonts: [
    {
      name: "Libre Franklin",
      styles: ["300", "400", "400i", "600", "600i"],
    },
    {
      name: "IBM Plex Mono",
      styles: ["400"],
    },
  ],
  headerFontFamily: ["Libre Franklin", "sans-serif"],
  bodyFontFamily: ["Libre Franklin", "sans-serif"],
  headerWeight: 600,
  boldWeight: 600,
  scaleRatio: 3,
})

export default typography
