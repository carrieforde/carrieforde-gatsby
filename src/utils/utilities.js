export const slugify = (slug) => slug.replace(/\s/g, '-').toLowerCase();
