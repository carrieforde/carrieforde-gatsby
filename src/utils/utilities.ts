export const slugify = (slug: string): string =>
  slug.replace(/\s/g, "-").toLowerCase();

export function getComponentKey(name: string, index: number): string {
  return `${slugify(name)}-${index}`;
}
