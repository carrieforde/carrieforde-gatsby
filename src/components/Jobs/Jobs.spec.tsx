import React from 'react';
import Jobs from './Jobs';
import { render } from '@testing-library/react';
import { mocked } from 'ts-jest/utils';
import { useStaticQuery } from 'gatsby';
import { jobs } from 'mocks/jobs';

jest.mock('gatsby');
jest.mock('gatsby-plugin-mdx', () => ({
  MDXRenderer: () => <div></div>,
}));

const useStaticQueryMock = mocked(useStaticQuery);

describe('Jobs', () => {
  useStaticQueryMock.mockReturnValue(jobs);

  it('should render the Jobs component', () => {
    const { container } = render(<Jobs />);
    expect(container).toMatchSnapshot();
  });
});
