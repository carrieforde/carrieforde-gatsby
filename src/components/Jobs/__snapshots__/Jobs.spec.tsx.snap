// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jobs should render the Jobs component 1`] = `
<div>
  <div
    class="tabs"
  >
    <div
      aria-label="Jobs"
      class="tabsList"
      role="tablist"
    >
      <button
        aria-controls="-panel"
        aria-selected="true"
        class="tabButton tabButtonIsActive"
        id="-tab"
        role="tab"
        tabindex="-1"
      />
      <button
        aria-controls="-panel"
        aria-selected="true"
        class="tabButton tabButtonIsActive"
        id="-tab"
        role="tab"
        tabindex="-1"
      />
      <button
        aria-controls="-panel"
        aria-selected="true"
        class="tabButton tabButtonIsActive"
        id="-tab"
        role="tab"
        tabindex="-1"
      />
      <button
        aria-controls="-panel"
        aria-selected="true"
        class="tabButton tabButtonIsActive"
        id="-tab"
        role="tab"
        tabindex="-1"
      />
    </div>
    <div
      aria-labelledby="-tab"
      class="tabPanel jobPanel"
      id="-panel"
      role="tabpanel"
      tabindex="0"
    >
      <div />
    </div>
    <div
      aria-labelledby="-tab"
      class="tabPanel jobPanel"
      id="-panel"
      role="tabpanel"
      tabindex="0"
    >
      <div />
    </div>
    <div
      aria-labelledby="-tab"
      class="tabPanel jobPanel"
      id="-panel"
      role="tabpanel"
      tabindex="0"
    >
      <div />
    </div>
    <div
      aria-labelledby="-tab"
      class="tabPanel jobPanel"
      id="-panel"
      role="tabpanel"
      tabindex="0"
    >
      <div />
    </div>
  </div>
</div>
`;
